### PR TITLE
[#445] Fixed scoping in try-with-resources

### PR DESCRIPTION
**Pull request by szarnekow**

### CODE REVIEW

File: org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/xbase/junit/evaluation/AbstractXbaseEvaluationTest.java [hunk 1]
```diff
@@ -2928,13 +2928,12 @@
 	 */
 	@Test @Ignore public void testTryWithResources_catch() throws Exception {
```
Comment (cdietrich) [inline org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/xbase/junit/evaluation/AbstractXbaseEvaluationTest.java]: btw why is here a ignore?

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/XbaseTypeComputer.java [hunk 2]
```diff
@@ -1166,17 +1166,18 @@ protected void _computeTypes(XTryCatchFinallyExpression object, ITypeComputation
 				caughtExceptions.add(referenceOwner.toLightweightTypeReference(catchClause.getDeclaredParam().getParameterType()));
 		}
 		
+		ITypeComputationState withExpectedExceptions = state.withExpectedExceptions(caughtExceptions);
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/XbaseTypeComputer.java]: do i see it correct we dont have a negative test here?

File: org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/xbase/junit/evaluation/AbstractXbaseEvaluationTest.java (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext.purexbase.tests/src/org/eclipse/xtext/xbase/junit/evaluation/AbstractXbaseEvaluationTest.java]: Xbase has no syntax for try with resource. It has to be enabled in sublangs for backwards compat reasons

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/XbaseTypeComputer.java (hunk 2 above)
Comment (szarnekow) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/XbaseTypeComputer.java]: I’ll add a linking test

**Pull request merged**